File: Projeto Didio Pizza/Didio Pizza Springboot/backend/src/main/java/com/pizzaria/didiopizza/repository/LojaRepository.java
package com.pizzaria.didiopizza.repository;

import java.util.Map;

import org.springframework.jdbc.core.JdbcTemplate;
import org.springframework.stereotype.Repository;

@Repository
public class LojaRepository {
    private final JdbcTemplate db;

    public LojaRepository(JdbcTemplate db) {
        this.db = db;
    }

    public Map<String, Object> buscarStatusLoja() {
        var l = db.queryForList("SELECT aberta FROM loja WHERE id = 1");
        return l.isEmpty() ? null : l.get(0);
    }

    public int alterarStatusLoja(boolean status) {
        return db.update("UPDATE loja SET aberta = ? WHERE id = 1", status);
    }
}

File: Projeto Didio Pizza/Didio Pizza Node/backend/src/repository/lojaRepository.js
import connection from './connection.js';

export async function buscarStatusLoja() {
    let comando = `
        SELECT aberta
        FROM loja
        WHERE id = 1
    `;

    const [linhas] = await connection.query(comando);

    return linhas[0];
}

export async function alterarStatusLoja(status) {
    let comando = `
        UPDATE loja
        SET aberta = ?
        WHERE id = 1
    `;

    const [resposta] = await connection.query(comando, [status]);

    return resposta.affectedRows;
}